src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts

```diff
@@ -0,0 +1,484 @@
+    "texture_depth_cube_array": WebGPUConstants.TextureViewDimension.CubeArray,
+    "texture_depth_multisampled_2d": WebGPUConstants.TextureViewDimension.E2d,
+    "texture_storage_1d": WebGPUConstants.TextureViewDimension.E1d,
+    "texture_storage_2d": WebGPUConstants.TextureViewDimension.E2d,
+    "texture_storage_2d_array": WebGPUConstants.TextureViewDimension.E2dArray,
+    "texture_storage_3d": WebGPUConstants.TextureViewDimension.E3d,
+    "texture_external": null
+};
+
+ /** @hidden */
+export class WebGPUShaderProcessorWGSL extends WebGPUShaderProcessor implements IShaderProcessor {
+
+    protected _attributesWGSL: string[];
+    protected _attributesDeclWGSL: string[];
+    protected _attributeNamesWGSL: string[];
+    protected _varyingsWGSL: string[];
+    protected _varyingsDeclWGSL: string[];
+    protected _varyingNamesWGSL: string[];
+
+    public shaderLanguage = ShaderLanguage.WGSL;
+    public uniformRegexp = /uniform\s+(\w+)\s*:\s*(.+)\s*;/;
+    public textureRegexp = /var\s+(\w+)\s*:\s*((array<\s*)?(texture_\w+)\s*(<\s*(.+)\s*>)?\s*(,\s*\w+\s*>\s*)?);/;
+    public noPrecision = true;
+    public removeCommentsBeforeProcessing = true;
+
+    protected _getArraySize(name: string, uniformType: string, preProcessors: { [key: string]: string }): [string, string, number] {
+        let length = 0;
+
+        const endArray = uniformType.lastIndexOf(">");
+        if (uniformType.indexOf("array") >= 0 && endArray > 0) {
+            let startArray = endArray;
+            while (startArray > 0 && uniformType.charAt(startArray) !== ' ' && uniformType.charAt(startArray) !== ',') {
+                startArray--;
+            }
+            const lengthInString = uniformType.substring(startArray + 1, endArray);
+            length = +(lengthInString);
+            if (isNaN(length)) {
+                length = +(preProcessors[lengthInString.trim()]);
+            }
+            while (startArray > 0 && (uniformType.charAt(startArray) === ' ' || uniformType.charAt(startArray) === ',')) {
+                startArray--;
+            }
+            uniformType = uniformType.substring(uniformType.indexOf("<") + 1, startArray + 1);
+        }
+
+        return [name, uniformType, length];
+    }
+
+    public initializeShaders(processingContext: Nullable<ShaderProcessingContext>): void {
+        this.webgpuProcessingContext = processingContext as WebGPUShaderProcessingContext;
+
+        this._attributesWGSL = [];
+        this._attributesDeclWGSL = [];
+        this._attributeNamesWGSL = [];
+        this._varyingsWGSL = [];
+        this._varyingsDeclWGSL = [];
+        this._varyingNamesWGSL = [];
+    }
+
+    public varyingProcessor(varying: string, isFragment: boolean, preProcessors: { [key: string]: string }, processingContext: Nullable<ShaderProcessingContext>) {
+        const varyingRegex = new RegExp(/\s*varying\s+(?:(?:highp)?|(?:lowp)?)\s*(\S+)\s*:\s*(.+)\s*;/gm);

```

Comment:
can those regexps be reused over classes? (seems to be the same as in  WebGPUShaderProcessorGLSL with a slight modification - (.+) instead of (\S+), so I also wonder if the change is needed).

Reply:
No, the regexp can be close to each other but the syntax in WGSL is different than the one in GLSL, so I prefer having separate regexp.